### PR TITLE
Generate more logs during the firstboot proccess.

### PR DESCRIPTION
**Pull request by SchnWalter**
Here's a an example log generated on my device:

```
1441229083: Running MIA init script: /system/etc/init.d/Y02firstboot
1441229083: Environment PATH variable: /sbin:/system/sbin:/system/bin:/system/xbin
1441229083: FirstBoot script found: /data/local/mia-firstboot.sh
1441229083: Waiting for com.android.systemui...
1441229083: - waiting, next try in 8s...
1441229091: - waiting, next try in 8s...
1441229099: - waiting, next try in 8s...
1441229107: - waiting, next try in 8s...
1441229115: - waiting, next try in 8s...
1441229123: - waiting, next try in 8s...
1441229132: - waiting, next try in 8s...
1441229140: - waiting, next try in 8s...
1441229148: - waiting, next try in 8s...
1441229156: - waiting, next try in 8s...
1441229164: - waiting, next try in 8s...
1441229172: - waiting, next try in 8s...
1441229180: - waiting, next try in 8s...
1441229188: - waiting, next try in 8s...
1441229196: - waiting, next try in 8s...
1441229204: - waiting, next try in 8s...
1441229212: - waiting, next try in 8s...
1441229220: - waiting, next try in 8s...
1441229228: com.android.systemui is running!
1441229235: Running script: /data/local/mia-firstboot.sh
1441229236: Updating UIDs in the orwall preferences file.
1441229236: Importing Superuser database with orWall uid pre-authorized.
1441229236: Generating SQL to populate the orWall NAT database.
1441229236: Using app data from file: /sdcard/misc/app-list-orwall.txt.
1441229236: - preparing SQL for org.fdroid.fdroid; uid: 10075
1441229236: - preparing SQL for info.guardianproject.orfox; uid: 10070
1441229236: - preparing SQL for com.android.providers.downloads; uid: 10006
1441229236: - preparing SQL for org.mariotaku.twidere; uid: 10076
1441229236: - preparing SQL for com.cyanogenmod.updater; uid: 10001
1441229236: - preparing SQL for com.csipsimple; uid: 10062
1441229236: - preparing SQL for com.fsck.k9; uid: 10063
1441229236: - preparing SQL for com.morlunk.mumbleclient; uid: 10064
1441229236: - preparing SQL for com.xabber.androiddev; uid: 10066
1441229236: - preparing SQL for de.schildbach.wallet; uid: 10067
1441229236: - preparing SQL for org.mozilla.firefox; uid:
1441229236: - preparing SQL for org.mozilla.fennec_fdroid; uid: 10077
1441229236: - preparing SQL for net.osmand.plus; uid: 10071
1441229237: - preparing SQL for fr.kwiatkowski.ApkTrack; uid: 10068
1441229237: - preparing SQL for indrora.atomic; uid: 10069
1441229237: Updating the orWall NAT database.
1441229237: Fix application data owner and group:
1441229237: - fixing ownership of com.android.inputmethod.latin; uid: 10039
1441229237: - fixing ownership of com.android.nfc; uid: 1027
1441229237: - fixing ownership of com.android.settings; uid: 1000
1441229237: - fixing ownership of com.xabber.androiddev; uid: 10066
1441229237: - fixing ownership of org.ethack.orwall; uid: 10074
1441229237: - fixing ownership of org.torproject.android; uid: 10081
1441229237: Settings.Secure.ADB_ENABLED -- Disable ADB by default.
1441229239: Settings.Secure.LOCATION_PROVIDERS_ALLOWED -- Limit the list of location providers that activities may access.
1441229240: Settings.Global.AUTO_TIME -- Do not update the date, time and time zone automatically from the network (NITZ).
1441229241: Settings.Global.AIRPLANE_MODE_ON -- Enable Airplane Mode and broadcast the event.
1441229244: Change hostname for subsequent reboots.
1441229245: Change hostname for current boot.
1441229248: Remove setup directory: /sdcard/misc
1441229248: Finished running script: /data/local/mia-firstboot.sh
1441229248: Removing firstboot script: /data/local/mia-firstboot.sh
1441229248: Finished running MIA init script: /system/etc/init.d/Y02firstboot
```

### CODE REVIEW

File: mia/templates/mia-default/archive/system/etc/init.d/Y02firstboot [hunk 1]
```diff
@@ -1,49 +1,50 @@
 #!/system/bin/sh
-# execute post-install script on First Boot
-# 01022012
+# Call `mia-firstboot.sh` if it exists.
+# @see https://github.com/CyanogenMod/android_vendor_cm/blob/cm-12.1/prebuilt/common/etc/init.d/90userinit
 
-FB_SCRIPT="/data/.firstboot.sh"
-TIMEOUT=5
+FB_SCRIPT="/data/local/mia-firstboot.sh"
```
Comment (patcon) [inline mia/templates/mia-default/archive/system/etc/init.d/Y02firstboot]: Let's stick with the community convention on the name/location :)
http://forum.xda-developers.com/showthread.php?t=1441378

**Comment by patcon** [comment]
Looks awesome! Just that one comment. (we didn't invent the firstboot thing ourselves)

**Comment by SchnWalter** [comment]
From what I have seen so far, I believe that we will need 2 scripts, one that runs a bit earlier in the boot process and one that runs later... that is why I have renamed things.

File: mia/templates/mia-default/archive/system/etc/init.d/Y02firstboot (hunk 1 above)
Comment (SchnWalter) [inline mia/templates/mia-default/archive/system/etc/init.d/Y02firstboot]: Actually, this is is the same location used for the CM `userinit.sh` and `userinit.d` scripts. And these scripts are very similar to the firstboot script, with the exception that the firstboot script is first waiting for the `com.android.systemui` process to finish.

https://github.com/CyanogenMod/android_vendor_cm/blob/cm-11.0/prebuilt/common/etc/init.d/90userinit
https://github.com/CyanogenMod/android_vendor_cm/blob/cm-12.1/prebuilt/common/etc/init.d/90userinit

**Comment by SchnWalter** [comment]
I've tested this half a dozen times, and it seems to work great... well it almost works great - there are a couple of glitches caused by newer apps, but the issues are unrelated to the changes in this PR.

see #107

**Comment by SchnWalter** [comment]
Just realized that this only works if it is installed over the default android. It won't work if you format the system partition and install CM; and then install the `mia-*.zip` files over a working "clean" CM (aka, no artifacts from the previous system)

**Comment by SchnWalter** [comment]
Looks like there was a regression in the latest Yamale version. See 23andMe/Yamale#22

I would say that we can merge this PR. There are still some issues but those should be tackled in other PR (404 error for the Flock app; FF crashing - caused by wrong lib files permissions)

**Pull request merged**